File: frontend/src/api/report.ts
import request from './request'

export interface TestReport {
  id: number
  scene_id: number
  status: string
  total_steps: number
  passed_steps: number
  failed_steps: number
  duration_ms?: number
  started_at: string
  finished_at?: string
}

export interface ReportStep {
  id: number
  report_id: number
  step_name: string
  step_type: string
  status: string
  request_data?: any
  response_data?: any
  assertions_result?: any[]
  error_message?: string
  duration_ms?: number
  created_at: string
}

export const reportApi = {
  list: (sceneId: number, limit = 50) => 
    request.get<TestReport[]>(`/reports/scene/${sceneId}?limit=${limit}`),
  get: (id: number) => request.get<TestReport>(`/reports/${id}`),
  getSteps: (reportId: number) => request.get<ReportStep[]>(`/reports/${reportId}/steps`),
  listAll: (projectId: number, limit = 100) =>
    request.get(`/reports?project_id=${projectId}&limit=${limit}`)
}

File: backend/app/schemas/report.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class RunSceneRequest(BaseModel):
    scene_id: int
    environment_id: Optional[int] = None


class TestReportResponse(BaseModel):
    id: int
    scene_id: int
    status: str
    total_steps: int
    passed_steps: int
    failed_steps: int
    duration_ms: Optional[int] = None
    started_at: datetime
    finished_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ReportStepResponse(BaseModel):
    id: int
    report_id: int
    step_name: str
    step_type: str
    status: str
    request_data: Optional[Dict[str, Any]] = None
    response_data: Optional[Dict[str, Any]] = None
    assertions_result: Optional[List[Dict[str, Any]]] = None
    error_message: Optional[str] = None
    duration_ms: Optional[int] = None
    created_at: datetime

    class Config:
        from_attributes = True


class RunResultResponse(BaseModel):
    report_id: int
    status: str
    message: str